src/Datadog.Trace.ClrProfiler.Managed/Models/HttpContextSpanIntegrationDelegate.cs

```diff
@@ -88,11 +88,11 @@ public override bool TryGetResourceName(out string resourceName)
         { // TODO: This will need to be moved out into a strategy to support extending to other integrations...
             var httpMethod = (GetHttpMethod() ?? Scope?.Span?.GetHttpMethod() ?? "GET").ToUpperInvariant();
 
-            var suffix = _httpContext.Request == null
-                             ? string.Empty
-                             : _httpContext.Request.Path ?? _httpContext.Request.RawUrl ?? string.Empty;
+            string suffix = _httpContext.Request?.Url == null
+                                ? string.Empty
+                                : UriHelpers.GetRelativeUrl(_httpContext.Request.Url, tryRemoveIds: true);
 
-            resourceName = string.Concat(httpMethod, " ", suffix.ToLowerInvariant()).Trim();
+            resourceName = $"{httpMethod} {suffix.ToLowerInvariant()}";
```

Comment:
Note that by removing `Trim()`, if `suffix` is an empty string then `resourceName` will have a trailing space, which we probably want to avoid.

Reply:
If suffix is empty, should we default to a slash?